purpleair-aqi.js

```diff
@@ -89,184 +128,191 @@ const levelAttributes = [
 }
 
-// Function to get the EPA adjusted PPM
-function computePM(data) {
-  let adj1 = parseInt(data.adj1, 10);
-  let adj2 = parseInt(data.adj2, 10);
-  let hum = parseInt(data.hum, 10);
-  let dataAverage = ((adj1 + adj2)/2);
-
-  // Apply EPA draft adjustment for wood smoke and PurpleAir
-  // from https://cfpub.epa.gov/si/si_public_record_report.cfm?dirEntryId=349513&Lab=CEMM&simplesearch=0&showcriteria=2&sortby=pubDate&timstype=&datebeginpublishedpresented=08/25/2018
-
-  return ((0.52 * dataAverage) - (.085 * hum) + 5.71);
-}
-
-// Function to get AQI number from PPM reading
+/**
+ * Get AQI number from PPM reading
+ *
+ * @param {number} pm
+ * @returns {number|'-'}
+ */
 function aqiFromPM(pm) {
-  if (pm > 350.5) {
-    return calcAQI(pm, 500.0, 401.0, 500.0, 350.5);
-  } else if (pm > 250.5) {
-    return calcAQI(pm, 400.0, 301.0, 350.4, 250.5);
-  } else if (pm > 150.5) {
-    return calcAQI(pm, 300.0, 201.0, 250.4, 150.5);
-  } else if (pm > 55.5) {
-    return calcAQI(pm, 200.0, 151.0, 150.4, 55.5);
-  } else if (pm > 35.5) {
-    return calcAQI(pm, 150.0, 101.0, 55.4, 35.5);
-  } else if (pm > 12.1) {
-    return calcAQI(pm, 100.0, 51.0, 35.4, 12.1);
-  } else if (pm >= 0.0) {
-    return calcAQI(pm, 50.0, 0.0, 12.0, 0.0);
-  } else {
-    return "-";
-  }
+  if (pm > 350.5) return calculateAQI(pm, 500.0, 401.0, 500.0, 350.5);
+  if (pm > 250.5) return calculateAQI(pm, 400.0, 301.0, 350.4, 250.5);
+  if (pm > 150.5) return calculateAQI(pm, 300.0, 201.0, 250.4, 150.5);
+  if (pm > 55.5) return calculateAQI(pm, 200.0, 151.0, 150.4, 55.5);
+  if (pm > 35.5) return calculateAQI(pm, 150.0, 101.0, 55.4, 35.5);
+  if (pm > 12.1) return calculateAQI(pm, 100.0, 51.0, 35.4, 12.1);
+  if (pm >= 0.0) return calculateAQI(pm, 50.0, 0.0, 12.0, 0.0);
+  return "-";
 }
 
-// Function that actually calculates the AQI number
-function calcAQI(Cp, Ih, Il, BPh, BPl) {
-  let a = (Ih - Il);
-  let b = (BPh - BPl);
-  let c = (Cp - BPl);
-  return Math.round((a/b) * c + Il);
+/**
+ * Calculate the AQI number
+ *
+ * @param {number} Cp
+ * @param {number} Ih
+ * @param {number} Il
+ * @param {number} BPh
+ * @param {number} BPl
+ * @returns {number}
+ */
+function calculateAQI(Cp, Ih, Il, BPh, BPl) {
+  const a = Ih - Il;
+  const b = BPh - BPl;
+  const c = Cp - BPl;
+  return Math.round((a / b) * c + Il);
 }
 
-// Calculates the AQI level based on
-// https://cfpub.epa.gov/airnow/index.cfm?action=aqibasics.aqi#unh
+/**
+ * Calculates the AQI level
+ * based on https://cfpub.epa.gov/airnow/index.cfm?action=aqibasics.aqi#unh
+ *
+ * @param {number|'-'} aqi
+ * @returns {LevelAttribute & { level: number }}
+ */
 function calculateLevel(aqi) {
-  let res = {
-    level: "0",
-    label: "Weird",
-    startColor: "white",
-    endColor: "white",
-    textColor: "black",
-    darkStartColor: "009900",
-    darkEndColor: "007700",
-    darkTextColor: "000000",
+  const level = Number(aqi) || 0;
+
+  const {
+    label = "Weird",
```

Comment:
This syntax means "if `label` is undefined in the object returned from `LEVEL_ATTRIBUTES.find`, then use `"weird"` as a default".